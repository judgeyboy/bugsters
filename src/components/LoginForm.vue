<template>
  <form @submit.prevent="onSubmit">
    <legend>Login</legend>
    <input type="text" placeholder="email" v-model="email">
    <input type="password" placeholder="password" v-model="password">
    <input type="submit" value="Login">
  </form>
</template>

<script>
export default {

  name: 'LoginForm',

  data () {
    return {
      email: '',
      password: ''
    }
  },

  methods: {
    onSubmit: function () {
      const loginDetails = {
        email: this.email,
        password: this.password
      }

      this.$store.dispatch('login', loginDetails)
    }
  }

}
</script>
