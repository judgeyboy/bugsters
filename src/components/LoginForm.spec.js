import Vue from 'vue'
import Vuex from 'vuex'
import { shallow } from 'vue-test-utils'
import LoginForm from './LoginForm'

Vue.use(Vuex)

describe('LoginForm component', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store({
      state: {},
      actions: {
        login: jest.fn()
      }
    })
  })

  it('snapshot', () => {
    const wrapper = shallow(LoginForm)
    const $html = wrapper.vm.$el.outerHTML
    expect($html).toMatchSnapshot()
  })

  it('should dispatch "login" with login details', () => {
    const wrapper = shallow(LoginForm, { store })
    const spy = jest.spyOn(wrapper.vm.$store, 'dispatch')

    const loginDetails = {
      email: 'duchess@spy-agency.com',
      password: 'DangerZone!'
    }

    wrapper.setData(loginDetails)

    const submitButton = wrapper.find('input[type=submit]')
    submitButton.trigger('click')

    expect(spy).toHaveBeenCalled()
    expect(spy).toHaveBeenCalledWith('login', loginDetails)
  })
})
