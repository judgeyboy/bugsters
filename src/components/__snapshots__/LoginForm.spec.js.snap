// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component snapshot 1`] = `"<form><legend>Login</legend> <input type=\\"text\\" placeholder=\\"email\\"> <input type=\\"password\\" placeholder=\\"password\\"> <input type=\\"submit\\" value=\\"Login\\"></form>"`;
